test/Test/Database/LSMTree/Internal/RunBuilder.hs

```diff
@@ -0,0 +1,58 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Test.Database.LSMTree.Internal.RunBuilder (tests) where
+
+import           Control.Monad (void)
+import           Control.Monad.Class.MonadThrow
+import           Data.Functor ((<&>))
+import           Database.LSMTree.Internal.Entry (NumEntries (..))
+import           Database.LSMTree.Internal.Paths (RunFsPaths (..))
+import qualified Database.LSMTree.Internal.RunBuilder as RunBuilder
+import qualified System.FS.API as FS
+import           System.FS.API (HasFS)
+import           System.FS.IO (HandleIO)
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+import           Test.Util.FS (withTempIOHasFS)
+
+tests :: TestTree
+tests = testGroup "Test.Database.LSMTree.Internal.RunBuilder" [
+      testProperty "prop_newInExistingDir" $ ioProperty $
+        withTempIOHasFS "prop_newInExistingDir" prop_newInExistingDir
+    , testProperty "prop_newInNonExistingDir" $ ioProperty $
+        withTempIOHasFS "prop_newInNonExistingDir" prop_newInNonExistingDir
+    , testProperty "prop_newTwice" $ ioProperty $
+        withTempIOHasFS "prop_newTwice" prop_newTwice
+    ]
+
+-- | 'new' in an existing directory should be succesfull.
+prop_newInExistingDir :: HasFS IO HandleIO -> IO Property
+prop_newInExistingDir hfs = do
+    let runDir = FS.mkFsPath ["a", "b", "c"]
+    FS.createDirectoryIfMissing hfs True runDir
+    try (RunBuilder.new hfs (RunFsPaths runDir 17) (NumEntries 0) 0) <&> \case
+      Left e@FS.FsError{} ->
+        counterexample ("expected a success, but got: " <> show e) $ property False
+      Right _ -> property True
```

Comment:
This seems like an individual test case, why are we running it as a property?

Reply:
The property is run only once because there are no arguments we have to generate, so it's isomorphic to `HUnit` test case